DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/README.md

```diff
@@ -0,0 +1,208 @@
+
+<img src="anr_ip.png" alt="anr_ip" width="900"/>
+
+To compute a given pixel, the `Vertical Kernel` must store previous rows (i.e., lines) of the input image. The technique to do so is shown in the image below. The pixels are streamed in from the pipe and used with pixels from previous rows to perform the 1D vertical window operation.
+
+<img src="vertical_kernel.png" alt="vertical_kernel" width="900"/>
+
+The logic for the `Horizontal Kernel`, shown below, is much simpler since it operates on a single row at a time.
+
+<img src="horizontal_kernel.png" alt="horizontal_kernel" width="900"/>
+
+To produce the input data and consume the output, we setup a full system as shown in the figure below. The `Input Kernel` reads input data from device memory and provides it to the ANR design via the input pipe. The `Output Kernel` reads the ANR design's output from the output pipe and writes it to device memory. The oneAPI host code then uses the output data to validate the accuracy of the ANR algorithm against a golden result using the [Peak signal-to-noise ratio (PSNR)](https://en.wikipedia.org/wiki/Peak_signal-to-noise_ratio).
+
+<img src="anr_system.png" alt="anr_system" width="800"/>
+
+### Quantized Floating-Point (QFP)
+Floating-point values consist of a sign bit, an exponent, and a mantissa. In this design, we take [32-bit single-precision](https://en.wikipedia.org/wiki/Single-precision_floating-point_format) floating values and convert them to quantized floating-point (QFP) values which use less bits. All of the QFPs in this design have 10 bits total, but use a different number for the exponent and mantissa. The purpose of this conversion is to be able to create lookup-table (LUT) read-only memories (ROMs) to approximate expensive 32-bit floating-point operations like an exponential (`exp(x)`) and inversion (`1/x`). Creating LUT ROMs for 32-bit floats would require `2^32*4 = 17GB` bytes of on-chip memory. However, if the float can be *quantized* to 10 bits, it requires only `2^10*4 = 4KB` of on-chip memory, at the expense of reduced precision.
+
+### Using Libraries
+In this design, we use the following generic libraries:
```

Comment:
Again, reusable/generic header files instead of libraries, just to avoid the heavily overloaded term 'libraries'?